.github/workflows/linux.yml

```diff
@@ -398,9 +398,141 @@ jobs:
+        run: |
+          mkdir -p nodejs
+          cp runtime/lib/intel64/* nodejs
+          cp runtime/3rdparty/tbb/lib/* nodejs
+          cp genai_node_addon.node nodejs
+          patchelf --set-rpath '$ORIGIN' libopenvino.so.2025.0.0
+        working-directory: ${{ env.INSTALL_DIR }}
+
+      - name: Pack Node.js bindings libs
+        run: tar -cvf - * | pigz > ${{ env.BUILD_DIR }}/genai_nodejs_bindings.tar.gz
+        working-directory: ${{ env.INSTALL_DIR }}/nodejs
+
+      - name: Upload Archive Package with Node.js bindings
+        if: ${{ always() }}
+        uses: actions/upload-artifact@b4b15b8c7c6ac21ea08fcf65892d2ee8f75cf882 # v4.4.3
+        with:
+          name: genai_nodejs_bindings
+          path: ${{ env.BUILD_DIR }}/genai_nodejs_bindings.tar.gz
+          if-no-files-found: 'error'
+
+      - name: Run javascript tests
+        working-directory: ${{ env.GENAI_REPO }}/src/js
+        run: |
+          cp -R ${{ env.INSTALL_DIR }}/nodejs bin
+          npm install
+          npm test
+
+      - name: Install genai-node samples dependencies
+        run: npm install
+        working-directory: ${{ env.GENAI_REPO }}/samples/js/chat_sample
+
+      - name: Run tests
+        run: npm test
+        env:
+          MODEL_PATH: ${{ env.GENAI_REPO }}/src/js/tests/models/Llama-3.2-3B-Instruct-openvino-8bit
+        working-directory: ${{ env.GENAI_REPO }}/samples/js/chat_sample
+
+    name: NodeJS Samples Tests
+    needs: [ genai_nodejs_bindings ]
+    timeout-minutes: 30
+    defaults:
+      run:
+        shell: bash
+    runs-on: ubuntu-20.04-16-cores
+    env:
+      INSTALL_DIR: ${{ github.workspace }}/ov
+      GENAI_REPO: ${{ github.workspace }}/openvino.genai
+      MODELS_DIR: ${{ github.workspace }}/models
+      JS_SRC_DIR: ${{ github.workspace }}/openvino.genai/src/js
+      JS_SAMPLES_DIR: ${{ github.workspace }}/openvino.genai/samples/js/chat_sample
+      BIN_DIR: ${{ github.workspace }}/openvino.genai/src/js/bin
+    steps:
+      - name: Clone openvino.genai
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          path: ${{ env.GENAI_REPO }}
+          submodules: recursive
+
+      - name: Download genai nodejs bindings archive
+        uses: actions/download-artifact@v4
+        with:
+          name: genai_nodejs_bindings
+          path: ${{ env.BIN_DIR }}
+
+      - name: Unpack genai nodejs bindings archive
+        run: tar -xzf genai_nodejs_bindings.tar.gz && rm genai_nodejs_bindings.tar.gz
+        working-directory: ${{ env.BIN_DIR }}
+
+      - name: Install nodejs dependencies
+        run: npm install
+        working-directory: ${{ env.JS_SRC_DIR }}
+
+      - name: Download model for tests
+        run: npm run test_setup
+        working-directory: ${{ env.JS_SRC_DIR }}
+
+      - name: Install genai-node samples dependencies
+        run: npm install
+        working-directory: ${{ env.JS_SAMPLES_DIR }}
+
+      - name: Run tests
```

Comment:
is it the same as `Run tests` in prev job? I would keep a single job for JS steps

Reply:
Left by mistake, removed